data/core.yaml

```diff
@@ -763,6 +763,8 @@ en:
     lock:
       suggestion: 'The "{label}" field is locked because there is a Wikidata tag. You can delete it or edit the tags in the "Tags" section.'
     display_name:
+      network_ref_name: "{network} {ref}: {name}"
+      ref_name: "{ref}: {name}"
```

Comment:
I sometimes encounter situations where it would be nice to also append `from=*`, `to=*`, and `via=*` for disambiguation – just like when the route has a `ref=*` instead of a `name=*`. Of course, the name might push all these extra details off out of view, but at least they’d appear in a tooltip as a last resort.

Reply:
👍 to adding a variant of `ref_from_to_via` which also includes the name at the end

Reply:
I’ve modified this PR to replace the short-circuit on `name` with something similar to what we do with `ref`. Now the label can combine the ref, name, and other details to avoid ambiguity. It only does this for routes; other things with names and refs, like indoor rooms, keep the same labels as before. Unfortunately, this change results in a combinatorial explosion of localizable strings, but hopefully they’re straightforward enough for translators to figure out.